text/204-golang-bindings.md

```diff
@@ -524,39 +680,57 @@ In Go: ([Go playground example](https://play.golang.org/p/T20xlddRo6A))
 ```go
 package greeter
 
-type GreeterIface interface {
-    Greet() string
+import "jsii"
+
+// The interface represents the class in the API.
+type Greeter interface {
+    Greet()       string
     GetGreeting() string
 }
 
-type Greeter struct {
-    Greeting string
+// The struct is the concrete implementation for the type. This is a JS object
+// proxy.
+type greeter struct {
+    // We need padding to ensure the struct is not 0-width, otherwise object
+    // identity is impossible to verify (if the struct occupies no memory, any
+    // object allocated right after an instance of it will share the exact same
+    // memory address).
+    _ byte // padding
 }
 
-func NewGreeter(message string) *Greeter {
-    return &Greeter{message}
+func NewGreeter(message string) Greeter {
+    g := &Greeter{}
+    // Creating the backing instance in the JS process
+    jsii.Create(g, "example.Greeter", []interface{}{message})
+    return g
 }
 
-func (g *Greeter) GetGreeting() string {
-    return g.Greeting
+func (g *greeter) GetGreeting() (result string) {
+    // Getting the property from the JS process
+    jsii.Get(g, "greeting", &result)
+    return
 }
 
-func (g *Greeter) Greet() string {
-    return fmt.Sprintf("Hello, %+s", g.Greeting)
+func (g *greeter) Greet() (result string) {
+    // Invoking the method in the JS process
+    jsii.Invoke(g, "greet", []interface{}{}, &result)
+    return
 }
 
 // static method
-func GreeterFoo() string {
-    return fmt.Sprintf("foo");
+func GreeterFoo() (result string) {
```

Comment:
How do we disambiguate this from a class named `GreeterFoo` (same problem we have with enums)?

Reply:
I guess we could name it `Greeter_Foo` here instead of `GreeterFoo`... This isn't super pretty but is probably the safest bet.

I actually haven't spent all that much time thinking about this case as part of this change.

Reply:
See my comment about enums. I think we want statics to look like enum members so underscore is the way to go.